hmrc/devops-tooling-build
# CIE-1476/Ensure 1hr min cached session duration

## What and Why

Container now set to assume roles for 2 hrs
and cache this.

Errors locally/Jenkins would occur as the cache
may have been valid up to 59 minutes, then a
long command run and the role assumption
expiring in the middle of the process.

This change ensures the cache is expired if
there is less than 1 hour remaining on the
cached session - i.e. it ensures a cached
session guarantees a minimum of 1 hr and a max
of 2.

As we are assuming for 2 hours, this maintains
parity with interface to user, that they can
expect to need to re-assume a role and enter an
MFA code once every hour per role. But there is
the enhancement that cached sessions are always
an hour, and their sessions can last UP TO 2 hrs
on average.

## How to review

Check code change / logic makes sense.

Ensure tests pass:
- make test/base
- make test/tf v=0.12.31
- make test/ami v=1.0.0

## Who can review

Anyone but @JamesLuaHMRC.
